preemptive_requests.py

```diff
@@ -1,26 +1,96 @@
 import json
-from qtpy import QtWidgets
+from string import Template
+
+from qtpy import QtCore, QtWidgets
 from pydm import Display
-from pydm.widgets import PyDMEmbeddedDisplay
+from PyQt5.QtGui import QTableWidgetItem, QIcon, QPixmap
+from fast_faults import VisibilityEmbedded
+
+
+class CustomTableWidgetItem(QTableWidgetItem):
+    """
+    Custom QTableWidgetItem to allow sorting items in a QTableWidget
+    based on the values from a PyDMEmbeddedDisplay widget.
+
+    Parameters
+    ----------
+    widget_type : QtWidgets
+        Type of widget, eg. `QtWidgets.QLabel`
+    widget_name : str
+        The name of the widget (object name).
+    """
+    def __init__(self, widget_type, widget_name, parent=None):
+        QTableWidgetItem.__init__(self, parent)
+        self._obj_name = widget_name
+        self._obj_type = widget_type
+
+    def __lt__(self, other):
+        """
+        Override the __lt__ to handle data sorting for rate.
+        """
+        # column 0 is where the embedded display widget data is at
+        column = 0
+        try:
+            other_widget = other.tableWidget().cellWidget(other.row(), column)
+            other_label = other_widget.embedded_widget.ui.findChild(
+                self._obj_type, str(self._obj_name))
+
+            widget = self.tableWidget().cellWidget(self.row(), column)
+            label = widget.embedded_widget.ui.findChild(
+                self._obj_type, str(self._obj_name))
+            other_value = ''.join(filter(str.isdigit, other_label.text()))
+            value = ''.join(filter(str.isdigit, label.text()))
```

Comment:
I need to figure out what to do with the PV entries that are disconnected here.... I could pretend that they are `maxsize` in  the `__lt__` method - and that would place them at the end of the table if sorting in ascending order, and on top if sorting the other way.... I am not sure if that's what we want though?

Here is a dummy example of how it would sort - the `PyDMLabel` value represents a PV that is disconnected:

not sorted:
![not_sorted](https://user-images.githubusercontent.com/62306310/110711362-482df580-81b4-11eb-9fed-7ea7542abb46.PNG)


sorted asc:
![sorted](https://user-images.githubusercontent.com/62306310/110711252-174dc080-81b4-11eb-95e3-b339b0433ee1.PNG)


sorted desc:
![sort_desc](https://user-images.githubusercontent.com/62306310/110711473-79a6c100-81b4-11eb-9571-f36fe961adfe.png)

Reply:
Currently, the disconnected PV entries in the fast fault tab simply disappear. Maybe that's what we want to do here too? I think keeping them maxsize is also a good solution.

Reply:
Yeah I am not sure why they don't disappear here, I think they should be disappearing because by default we're trying to filter by `ca://${P}${ARBITER}:AP:Entry:${POOL}:Live_RBV` - I will check this

Reply:
Regardless, I think the widgets are still there but they will not be visible, so if we have some disconnected PVs like we do here (for the `000:Live_RBV` PVs at least) - we're still going to add them to the Tab containers anyway, but they will not be visible.

I think it is still safer to still think that we could possibly have values in the PyDMLabels that are not necessarily numbers, and assume those to have maxsize....